plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -6268,4 +6337,24 @@ private List<Long> getSnapshotsIdsByCreationOrder(String tableName)
                 .map(row -> (Long) row.getField(idField))
                 .collect(toList());
     }
+
+    private QueryId getQueryIdFromLatestSnapshot(String tableName)
+    {
+        String findLatestQueryId = format("SELECT json_extract_scalar(CAST(SUMMARY AS JSON), '$.%s') FROM \"%s$snapshots\" ORDER BY committed_at DESC LIMIT 1", TRINO_QUERY_ID_NAME, tableName);
+        return getQueryRunner().execute(findLatestQueryId)
+                .getOnlyColumn()
+                .map(queryId -> QueryId.valueOf((String) queryId))
+                .findFirst()
+                .orElseThrow(() -> new IllegalStateException("Table " + tableName + " has zero snapshots or does not have the Trino query ID in its snapshot summary."));
+    }
+
+    private void executeAndAssertQueryIdStored(String tableName, String sql)
```

Comment:
Let's check here both the queryId and the `trino_query_id` and `trino_user`

Reply:
Done, checking both now